ci-operator/step-registry/openshift-extended/test/openshift-extended-test-commands.sh

```diff
@@ -224,6 +224,30 @@ oc wait nodes --all --for=condition=Ready=true --timeout=15m
 oc wait clusteroperators --all --for=condition=Progressing=false --timeout=15m
 oc get clusterversion version -o yaml || true
 
+# Get the KUBEADMIN_REMOVED environment variable (or set a default false if missing)
+KUBEADMIN_REMOVED="${KUBEADMIN_REMOVED:-false}"
+# Function to delete the kubeadmin secret if KUBEADMIN_REMOVED is "true"
+function remove_admin_user() {
```

Comment:
@dpuniaredhat 
```shell
# Get the KUBEADMIN_REMOVED environment variable (or set a default false if missing)
KUBEADMIN_REMOVED="${KUBEADMIN_REMOVED:-false}"
# Function to delete the kubeadmin secret if KUBEADMIN_REMOVED is "true"
function remove_admin_user() {
  if [[ "$KUBEADMIN_REMOVED" == "true" ]]; then
    echo "KUBEADMIN_REMOVED is set to 'true'. Deleting kubeadmin secret..."
    oc delete secrets kubeadmin -n kube-system

    if test -f "${SHARED_DIR}/nested_kubeconfig"; then
      echo "KUBEADMIN_REMOVED is set to 'true'. Deleting kubeadmin for GUEST Cluster"
      HOSTED_CLUSTER_NAME=$(oc get hostedclusters -n clusters -o jsonpath='{.items[0].metadata.name}')
      oc delete secret/kubeadmin-password -n "clusters-${HOSTED_CLUSTER_NAME}"
    fi
    # Check if deletion was successful and provide feedback
    if [[ $? -eq 0 ]]; then
      echo "Kubeadmin secret deleted successfully."
    else
      echo "Error deleting kubeadmin secret. Please check manually."
    fi
  else
    echo "KUBEADMIN_REMOVED not set to 'true'. Skipping deletion."
  fi
}
```
should be 
```shell
function remove_admin_user() {
    if [[ "$KUBEADMIN_REMOVED" == "true" ]]; then
        echo "KUBEADMIN_REMOVED is set to 'true'. Deleting kubeadmin secret..."
        ret_delete_admin=0

        // it is hosted cluster only for workflow cucushift-installer-rehearse-aws-ipi-ovn-hypershift-guest
        // it means you only see hosted cluster in testing
        if test -f "${SHARED_DIR}/nested_kubeconfig" && diff -q "${SHARED_DIR}/nested_kubeconfig" "${SHARED_DIR}/kubeconfig" >/dev/null; then
            HOSTED_CLUSTER_NAME=$(oc --kubeconfig="${SHARED_DIR}/mgmt_kubeconfig" get hostedclusters -n clusters -o jsonpath='{.items[0].metadata.name}' || ret_delete_admin=$?)
            if [ "W${ret_delete_admin}W" != "W0W" ]; then
                echo "fail to get HOSTED_CLUSTER_NAME"
                if [ "W${FORCE_SUCCESS_EXIT}W" == "WnoW" ]; then
                    echo "do not force success exit"
                    exit 1
                fi
                echo "force success exit"
                exit 0
            fi
            oc --kubeconfig="${SHARED_DIR}/mgmt_kubeconfig" delete secret/kubeadmin-password -n "clusters-${HOSTED_CLUSTER_NAME}" || ret_delete_admin=$?
            if [ "W${ret_delete_admin}W" != "W0W" ]; then
                echo "fail to delete kubeadmin-password in mgmt clusger for hosted cluster"
                if [ "W${FORCE_SUCCESS_EXIT}W" == "WnoW" ]; then
                    echo "do not force success exit"
                    exit 1
                fi
                echo "force success exit"
                exit 0
            fi
            echo "Kubeadmin secret deleted successfully for hosted cluster."
            return
        fi

        // it is mgmt cluster of hypershift or non-hypershift cluster (normal cluster)
        oc --kubeconfig="${SHARED_DIR}/kubeconfig" delete secrets kubeadmin -n kube-system || ret_delete_admin=$?
        if [ "W${ret_delete_admin}W" != "W0W" ]; then
            echo "fail to delete kubeadmin-password in mgmt clusger or non-hypershift cluster"
            if [ "W${FORCE_SUCCESS_EXIT}W" == "WnoW" ]; then
                echo "do not force success exit"
                exit 1
            fi
            echo "force success exit"
            exit 0
        fi
        echo "Kubeadmin secret deleted successfully for mgmt cluster or non-hypershift cluster."

    else
        echo "KUBEADMIN_REMOVED not set to 'true'. Skipping deletion."
    fi
}
```

the modified:
1, indent aligned
2, for cluster created by workflow cucushift-installer-rehearse-aws-ipi-ovn-hypershift-guest, the kubeconfi is already pointed hosted cluster per step cucushift-hypershift-extended-enable-guest, so your current logis is not correct. 
3, because of `set -o errexit`, we need to controll return code by ourslvle. or else the workflow will stop.
    so, modify code to control it.

I do not test it, but the overall logic should be correct.

you could test it by running 
1, your new change two jobs
2, one existing non-hypershift job
3, one existing hypershift job

see all these 4 jobs run as expected.

Reply:
As confirmed by dev as per designed, The kubeadmin password is automatically removed as soon as an identity provider is configured for the hosted cluster. So we do't need to do any changes for hypershift.
https://issues.redhat.com/browse/OCPBUGS-32276